lib/msal-node/docs/errors.md

```diff
@@ -0,0 +1,171 @@
+# Errors
+
+***
+
+**[ClientAuthErrors](#Clientautherrors)**
+
+1.[endpoints_resolution_error](#endpoints_resolution_error)
+
+2.[openid_config_error](#openid_config_error)
+
+3.[invalid_assertion](#invalid_assertion)
+
+4.[request_cannot_be_made](#request_cannot_be_made)
+
+5.[invalid_client_credential](#invalid_client_credential)
+
+**[ClientConfigurationErrors](#ClientConfigurationErrors)**
+
+1.[empty_url_error](#empty_url_error)
+
+2.[url_parse_error](#url_parse_error)
+
+**[Other](#other)**
+
+## ClientAuthErrors
+
+### Endpoints_resolution_error
+
+**Error Message**: Error: could not resolve endpoints. Please check network and try again.
+
+This error is thrown under the following scenarios:
+
+1. When a wrong authority endpoint is provided. Details: ClientAuthError: `openid_config_error`: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. 
+
+Ensure that the authority provided matches the authority in your azure app registration.
+```javascript
+const config = {
+    auth: {
+        authority: "https://login.microsoftonline.com/<provide-authority-that-matches-your-app-registration>"
+        // ...   
+    }
+}
+```
+
+2. When there is no internet connectivity. Details: ClientConfigurationError: `untrusted_authority`: The provided authority is not a trusted authority. Please include this authority in the knownAuthorities config parameter. It is also thrown when no authority is appended to the authority endpoint.
+    
+
+### Invalid_assertion
+
+**Error Message**: Client assertion must meet requirements described in https://tools.ietf.org/html/rfc7515. 
+
+This error is thrown when either the certificate thumbprint or public key is not provided or commented out.
+
+❌ The following example will throw this error because `thumbprint` is not provided:
+
+```javascript
+auth: {
+        //...
+        clientCertificate: {
+            thumbprint: "", // can be obtained when uploading certificate to Azure AD
+            privateKey: privateKey,
+        }
+    }
+```
+
+❌ The following example will throw this error because `publicKey` is not provided:
+
+```javascript
+auth: {
+        //...
+        clientCertificate: {
+            thumbprint: "YOUR_THUMBPRINT", // can be obtained when uploading certificate to Azure AD
+            privateKey: "",
+        }
+    }
+```
+
+
+### Request_cannot_be_made
+
+**Error message**: Token request cannot be made without authorization code or refresh token.
+
+This error is thrown when no authority is provided for auth flows such as auth-code-with-certs. 
```

Comment:
This sounds like a bug if you're experiencing this when not providing authority. MSAL should be using the default.